### ISSUE TITLE
`Error: Unsupported language version '0.8.27'`

### ISSUE THREAD

**Issue by pcaversaccio** [OpenZeppelin/openzeppelin-upgrades#1074]
For repro use my repo here: https://github.com/pcaversaccio/hardhat-project-template-ts

I updated the Solidity version to `0.8.27` via https://github.com/pcaversaccio/hardhat-project-template-ts/commit/f44a0350d47e26a56666a175d2abf9559e7eac0b.

The hardhat tests currently fail due to your package not supporting the Solidity version `0.8.27`: `npx hardhat test`:

```console
An unexpected error occurred:
Error: Unsupported language version '0.8.27'.
    at tryRemoveNonStructNatSpec (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+upgrades-core@1.37.0/node_modules/@openzeppelin/upgrades-core/src/utils/make-namespaced.ts:[18](https://github.com/blokk-studio/hardhat-project-template-ts/actions/runs/10752994779/job/29821936631#step:10:19)3:22)
    at makeNamespacedInput (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+upgrades-core@1.37.0/node_modules/@openzeppelin/upgrades-core/src/utils/make-namespaced.ts:158:28)
    at OverriddenTaskDefinition._action (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/@openzeppelin+hardhat-upgrades@3.2.1_@nomicfoundation+hardhat-ethers@3.0.8_ethers@6.13.2_buff_b6fk7igjaxyvtwxdu22yysbakm/node_modules/@openzeppelin/hardhat-upgrades/src/index.ts:102:31)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Environment._runTaskDefinition (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:351:14)
    at async Environment.run (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.[22](https://github.com/blokk-studio/hardhat-project-template-ts/actions/runs/10752994779/job/29821936631#step:10:23).10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:184:14)
    at async SimpleTaskDefinition.action (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/builtin-tasks/compile.ts:1024:37)
    at async Environment._runTaskDefinition (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:351:14)
    at async Environment.run (/home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/internal/core/runtime-environment.ts:184:14)
    at async /home/runner/work/hardhat-project-template-ts/hardhat-project-template-ts/node_modules/.pnpm/hardhat@2.22.10_bufferutil@4.0.8_ts-node@10.9.2_@types+node@22.5.4_typescript@5.5.4__typescri_7iajnuxsvogcbxkb6ol3j2urla/node_modules/hardhat/src/builtin-tasks/compile.ts:464:28 {
  code: 'GenericFailure'
}
 ELIFECYCLE  Command failed with exit code 1.
Error: Process completed with exit code 1.
```

I don't use your package at all, but it's required by the `@tenderly/hardhat-tenderly` package unfortunately. I haven't debugged what is triggering exactly this issue, but would appreciate a patch :)

**Comment by ericglau** [comment]
Thanks for reporting.  This occurs because the Slang dependency introduced in https://github.com/OpenZeppelin/openzeppelin-upgrades/pull/1051 (in release [@openzeppelin/upgrades-core@1.35.0](https://github.com/OpenZeppelin/openzeppelin-upgrades/releases/tag/%40openzeppelin%2Fupgrades-core%401.35.0)) doesn't yet support `0.8.27`.  We will fix this to avoid this code path for versions that Slang does not recognize.

**Comment by ericglau** [comment]
This is fixed in [@openzeppelin/upgrades-core@1.37.1](https://github.com/OpenZeppelin/openzeppelin-upgrades/releases/tag/%40openzeppelin%2Fupgrades-core%401.37.1)
